zcash_client_sqlite/src/lib.rs

```diff
@@ -631,76 +628,100 @@ impl<P: consensus::Parameters> WalletWrite for WalletDb<rusqlite::Connection, P>
         d_tx: DecryptedTransaction<AccountId>,
     ) -> Result<(), Self::Error> {
         self.transactionally(|wdb| {
-            let tx_ref = wallet::put_tx_data(wdb.conn.0, d_tx.tx, None, None)?;
+            let tx_ref = wallet::put_tx_data(wdb.conn.0, d_tx.tx(), None, None)?;
 
             let mut spending_account_id: Option<AccountId> = None;
-            for output in d_tx.sapling_outputs {
-                match output.transfer_type {
+            for output in d_tx.outputs() {
+                match output.transfer_type() {
                     TransferType::Outgoing | TransferType::WalletInternal => {
-                        let value = output.note.value();
-                        let recipient = if output.transfer_type == TransferType::Outgoing {
-                            Recipient::Sapling(output.note.recipient())
+                        let recipient = if output.transfer_type() == TransferType::Outgoing {
+                            match output.note() {
+                                Note::Sapling(n) => {
+                                    //TODO: Recover the UA, if possible.
```

Comment:
Open an issue for this.

Reply:
https://github.com/zcash/librustzcash/issues/1236